lib/api/profile.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	. "github.com/projectcalico/libcalico/lib/api/unversioned"
+)
+
+type ProfileMetadata struct {
+	ObjectMetadata
+	Labels map[string]string `json:"labels,omitempty" validate:"omitempty,labels"`
+}
+
+type ProfileSpec struct {
```

Comment:
I think all these structs are our public API definition and should probably have detailed comments to describe what each struct is, and how each field behaves.